### PR TITLE
auto type conversion

### LINKED ISSUE #4269: Default args in `StableDiffusionControlNetPipeline.check_inputs` causes TypeError
### Describe the bug

Default value of `control_guidance_start` and `control_guidance_end` in `StableDiffusionControlNetPipeline.check_inputs` function causes `TypeError: object of type 'float' has no len()`

### Reproduction

```
from diffusers import (
   StableDiffusionControlNetPipeline,
   ControlNetModel
)
import torch
from PIL import Image

controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny", torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained('runwayml/stable-diffusion-v1-5', controlnet=controlnet)

pipe.check_inputs('', Image.new('RGB',(512,512)), callback_steps=1)

```

### Logs

```shell
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
Traceback (most recent call last):
  File "bug.py", line 11, in <module>
    pipe.check_inputs('',Image.new('L',(512,512)),callback_steps=1)
  File "/home/neil/anaconda3/envs/reference_only/lib/python3.8/site-packages/diffusers/pipelines/controlnet/pipeline_controlnet.py", line 598, in check_inputs
    if len(control_guidance_start) != len(control_guidance_end):
TypeError: object of type 'float' has no len()
```

### System Info

- `diffusers` version: 0.18.2
- Platform: Linux-5.15.0-1037-gcp-x86_64-with-glibc2.17
- Python version: 3.8.16
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.30.2
- Accelerate version: 0.18.0
- xFormers version: 0.0.17
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@patrickvonplaten

### PR DESCRIPTION
**Pull request by isNeil**
Default value of `control_guidance_start` and `control_guidance_end` in `StableDiffusionControlNetPipeline.check_inputs` function causes `TypeError: object of type 'float' has no len()`

Proposed fix: 
Convert `control_guidance_start` and `control_guidance_end` to type list if float

# What does this PR do?

Fixes #4269

@holwech

### CODE REVIEW

File: src/diffusers/pipelines/controlnet/pipeline_controlnet.py [hunk 1]
```diff
@@ -554,6 +554,12 @@ def check_inputs(
         else:
             assert False
 
+        if isinstance(control_guidance_start, float):
+            control_guidance_start = [control_guidance_start]
+            
+        if isinstance(control_guidance_end, float):
+            control_guidance_end = [control_guidance_start]
```
Comment (isNeil) [inline src/diffusers/pipelines/controlnet/pipeline_controlnet.py:561]:
```suggestion
            control_guidance_end = [control_guidance_end]
```

**Review by isNeil — COMMENTED** [review]
typo

File: src/diffusers/pipelines/controlnet/pipeline_controlnet.py [hunk 2]
```diff
@@ -554,6 +554,12 @@ def check_inputs(
         else:
             assert False
 
+        if isinstance(control_guidance_start, float):
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/controlnet/pipeline_controlnet.py:557]:
```suggestion
        if not isinstance(control_guidance_start, (tuple, list)):
```

File: src/diffusers/pipelines/controlnet/pipeline_controlnet.py [hunk 3]
```diff
@@ -554,6 +554,12 @@ def check_inputs(
...
+            control_guidance_start = [control_guidance_start]
+            
+        if isinstance(control_guidance_end, float):
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/controlnet/pipeline_controlnet.py:560]:
```suggestion
        if not isinstance(control_guidance_end, (tuple, list)):
```

**Review by patrickvonplaten — COMMENTED** [review]
Bit safer to just check for non list or tuple (to also catch ints and tensors, ...)

**Comment by isNeil** [comment]
> Bit safer to just check for non list or tuple (to also catch ints and tensors, ...)

Yep, makes sense

**Comment by patrickvonplaten** [comment]
Thanks!

**Pull request merged**